lib/classes/Service.js

```diff
@@ -46,73 +47,36 @@ class Service {
       return BbPromise.resolve();
     }
 
-    // List of supported service filename variants.
-    // The order defines the precedence.
-    const serviceFilenames = [
-      'serverless.yaml',
-      'serverless.yml',
-      'serverless.json',
-      'serverless.js',
-    ];
-
-    const serviceFilePaths = _.map(serviceFilenames, filename => path.join(servicePath, filename));
-    const serviceFileIndex = _.findIndex(serviceFilePaths,
-      filename => this.serverless.utils.fileExistsSync(filename)
-    );
-
-    // Set the filename if found, otherwise set the preferred variant.
-    const serviceFilePath = serviceFileIndex !== -1 ?
-      serviceFilePaths[serviceFileIndex] :
-      _.first(serviceFilePaths);
-    const serviceFilename = serviceFileIndex !== -1 ?
-      serviceFilenames[serviceFileIndex] :
-      _.first(serviceFilenames);
-
-    if (serviceFilename === 'serverless.js') {
-      return BbPromise.try(() => {
-        // use require to load serverless.js file
-        // eslint-disable-next-line global-require
-        const configExport = require(serviceFilePath);
-        // In case of a promise result, first resolve it.
-        return configExport;
-      }).then(config => {
-        if (!_.isPlainObject(config)) {
-          throw new Error('serverless.js must export plain object');
-        }
-
-        return that.loadServiceFileParam(serviceFilename, config);
-      });
-    }
-
-    return that.serverless.yamlParser
-      .parse(serviceFilePath)
-      .then((serverlessFileParam) =>
-        that.loadServiceFileParam(serviceFilename, serverlessFileParam)
-      );
+    return BbPromise.all([
+      serverlessConfigFileUtils.getServerlessConfigFilePath(this.serverless),
+      serverlessConfigFileUtils.getServerlessConfigFile(this.serverless),
+    ]).then(that.loadServiceFileParam.bind(that));
   }
 
-  loadServiceFileParam(serviceFilename, serverlessFileParam) {
+  loadServiceFileParam([serviceFilename, serverlessFileParam]) {
```

Comment:
It'll be nicer to leave original signature (seems more natural), in above `then` it can be constructed as:

```javascript
args => this.loadServiceFileParam(...args)
```

_(I would also minimize internal API changes, to avoid risk of breaking some adventurous user plugins)_